db.go

```diff
@@ -397,6 +393,19 @@ func (db *DB) close() error {
 	return nil
 }
 
+func (db *DB) ensureFreelist() *freelist {
```

Comment:
Can you explain more how this lazy load helps?

It appears that you're trading

* an eager read (that will happen eventually anyway?)
* a regular pointer load
* straightforward code

for

* a lazy read
* an atomic pointer load
* less obvious code

I'd like to see more detail about the benefits to understand why this is worth doing.